src/pyspiffe/svid/jwt_svid.py

```diff
@@ -1,19 +1,118 @@
+from pyspiffe.svid.jwt_svid_validator import JwtSvidValidator
+
+
+INVALID_INPUT_ERROR = 'Invalid input: {}.'
 
 
 class JwtSvid(object):
     """
-    Represents a SPIFFE X.509 SVID.
+    Represents a SPIFFE JWT SVID as defined in the SPIFFE standard.
+    See <a href="https://github.com/spiffe/spiffe/blob/master/standards/JWT-SVID.md">https://github.com/spiffe/spiffe/blob/master/standards/JWT-SVID.md</a>.
+
+    Attributes:
+        spiffeId (SpiffeId): token spiffeId.
+        audience (List): audience claim.
+        expiry (datetime): date and time in UTC specifing expiry date.
+        claims (Dictionary): key-value pairs with all the claims present in the token.
+        token (str): encoded token.
+    """
 
-    Contains a SPIFFE ID, a private key and a chain of X.509 certificates.
+    def __init__(
+        self, spiffeId: SpiffeId, audience: [], expiry: datetime, claims: {}, token: str
+    ) -> None:
+        self.spiffeId = spiffeId
+        self.audience = audience
+        self.expiry = expiry
+        self.claims = claims
+        self.token = token
 
-    :param spiffe_id: SPIFFE ID of the X509-SVID.
-    :param private_key: Audience is the intended recipients of JWT-SVID as present in the 'aud' claim.
-    :param chain: Expiration time of JWT-SVID as present in 'exp' claim.
     """
+    Parses and validates a JWT-SVID token and returns an instance of a JwtSvid with a SPIFFE ID parsed from the 'sub', audience from 'aud',
+    and expiry from 'exp' claim. The JWT-SVID signature is not verified.
+
+    Args:
+        token(str): a token as a string that is parsed and validated.
+        param audience(List): audience as a list of strings used to validate the 'aud' claim.
+
+    Returns:
+        an instance of JwtSvid with a SPIFFE ID parsed from the 'sub', audience from 'aud', and expiry
+        from 'exp' claim.
+
+    Raises:
+        ValueError: when the token is blank or cannot be parsed, in case header is not specified,
+        InvalidAlgorithmError: in case specified 'alg' is not supported as specified by the SPIFFE standard.
+        InvalidTypeError: in case 'typ' is present in header but is not set to 'JWT' or 'JOSE'.
+        InvalidClaimError: in case a required claim is not present in payload or expected_audience is not a subset of audience_claim.
+        TokenExpiredError: in case token is expired.
+        ValueError: in case expected_audience is empty or if the string spiffe_id doesn't comply the SPIFFE standard.
+    """
+
+    @classmethod
+    def parse_insecure(cls, token: str, expected_audience: []) -> 'JwtSvid':
+        if not token:
+            raise ValueError(INVALID_INPUT_ERROR.format('token cannot be empty'))
+        token_header = jwt.get_unverified_header(token)
+        validator = JwtSvidValidator()
+        validator.validate_header(token_header)
+        claims = jwt.decode(token, options={'verify_signature': False})
+        validator.validate_claims(claims, expected_audience)
```

Comment:
Not sure if is possible based on the uses cases but in the scenario that `jwt.any_method()` raises and exception, is it "excepted" at some point and translated to one of the "local exceptions" or is just raised?

Reply:
That is a good point, Andres! I will make the change to catch it and translate to a "local" error-exception.

Reply:
Done!